### ISSUE TITLE
Use nntrainer model for feature extraction

### ISSUE THREAD

**Issue by klmntoth** [nnstreamer/nntrainer#927]
Hello!
I would like to use nntrainer as a feature extractor, like how tflite is used in the KNN example. But calling forwaring/inference uses the automatically added loss layer. Right now after calling inference i'm using `model.getFlatGraph().back()->getOutputs() ...` to get the values from the last layer added by me. Is there a better way to do this?

**Comment by kparichay** [comment]
Hello
There are multiple ways to extract the features from a pre-trained model - 

- As you described, running the `inference()` method and then `model.getFlatGraph().back()->getOutputs()`. However, this becomes complicated when extracting features from other layers.

Unfortunately, the current implementation of NNTrainer always attached a loss to the end of the model. With the loss attached, the inference includes the affect of the loss. In order to extract the features manually, there is no better way for now. 

Soon we will be supporting extracting nntrainer trained model with TfLite compatible format #899 and then, you can directly use your trained model for feature extraction.

If the extracted features are going to be used later for more processing, you can directly use the model till the last layer as a backbone for your final task.
If the goal is just to extract the features, there is another hack you can try meanwhile is to use a loss (MSE in this case) for the inference model which does not affect the output of the layer. The inference model can use a submodel of the full training model as a backbone, and then use the features correspondingly.

**Comment by kparichay** [comment]
We will add support for allowing no loss in the model, in order to facilitate your requirement.
The PR for this will be up soon.

**Comment by kparichay** [comment]
Please check #934 which allows setting loss as `none` or skipping adding loss to a model.
You can create two model definition files (ini), one for training and another for feature extraction (consider using backbone while specifying desired output layer), load the trained model binary file in the model created for feature extraction.

Please let me know if you have more questions on this.

**Comment by klmntoth** [comment]
Can you please give me an example of an ini file for feature extraction? I tried to make one with `loss=none` and using my original ini as backbone, but i'm getting way different values then using the method i posted before. Also i tried to simply use `setProperty({"loss=none"})` from the code using my originial ini, but i'm getting the same different values as with the second ini.

**Comment by kparichay** [comment]
As you noticed, using the method you posted before vs using `loss=none` can give different results in two scenarios:
1. Last layer is softmax and loss is cross-entropy
2. Last layer is sigmoid and loss is cross-entropy
For numerical stability, these two layers are merged together as a single operation. So, when you use your method, `model.getFlatGraph().back()->getOutputs()`, this would return the output of the model without softmax/sigmoid.
However, with `loss=none`, the output would include softmax/sigmoid operation.

I will add a few examples of ini file for feature extraction based on different loss and last layer activations soon.

**Comment by kparichay** [comment]
#934 has been updated to provide a `mnist_inf.ini` as an example to create a model without loss.
If you want to even remove the last layer, you can provide `InputLayer` and `OutputLayer` as arguments to `backbone` to select a submodel for feature extraction (refer ) or write the model in the `ini` file without the required layers/features (check line 63) as shown in comments in `mnist_inf.ini`.
